packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2300,12 +2300,50 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     }
   }
 
+  Set<SemanticsAction> _mutedSemanticsActions = <SemanticsAction>{};
+
+  /// Update the set of [SemanticsAction] that will be ignore when constructing
+  /// semantics action handlers.
+  ///
+  /// The muted [SemanticsAction] will apply to all the children in subtree.
+  void updateMutedSemanticsActions(Set<SemanticsAction> actions) {
```

Comment:
Do we have a use case were we only need to block selective actions? Or is it all-or-nothing?

Reply:
If we go the down the selective route - how do we deal with newly added semantics actions that people didn't know about when writing their render objects that use this functionality?

Reply:
I believe we might want to do selective route for example we are not likely to block
SemanticsAction.showOnScreen in ignoring pointer widget.

For newly added semantics actions, I really can't think of a comprehensive way to improve this. we can try to group the actions into different subgroup, for example, we can have a subgroup called PointerActions that will include all the pointer related semantics action, and the the renderobject can directly mute the subgroup instead individual actions. The person who add the new semantics action need to also include the action into the correct subgroup.

Reply:
showOnScreen is kind of a special SemanticsAction since it is not directly user-triggered. It would probably be good to have a way to block all user-initiated actions.

Having said that, showOnScreen as a semantics action is really only relevant if the node is scrollable. And scrolling is one of the actions we are blocking, with ignorepointer, right?